### DISCUSSION TITLE
Naming convention enforcement?

### DISCUSSION THREAD

**Discussion by jeremypng** [netbox-community/netbox#6854] [Q&A] [answered]
Does anyone have an example of using middleware in a plugin to enforce a naming convention?

**Comment by jeremystretch** [comment, marked as answer]
I don't think that would work very well, but check out the new custom validation feature coming in v3.0 (#5963): https://netbox.readthedocs.io/en/feature/customization/custom-validation/

**Reply by ryanmerolle** [reply]
Before custom validation, I personally used reports to audit the hostnames vs a standard.  Each device role has a standard defined.

Here is a super simple example: https://github.com/netbox-community/reports/blob/master/reports/dcim-reports/CheckDeviceNaming.py
